posts/2024-05-08_tlg_catalog_webr/tlg_catalog_webr.qmd

```diff
@@ -0,0 +1,70 @@
+---
+title: "TLG Catalog 🤝 WebR"
+author:
+  - name: Pawel Rucki
+description: "Introducing WebR to TLG Catalog: A Game Changer for Interactive Learning"
+date: "2024-05-08"
+# please do not use any non-default categories.
+# You can find the default categories in the repository README.md
+categories: [TLG, Shiny]
+# feel free to change the image
+image: "tlg_catalog.png"
+
+---
+
+<!--------------- typical setup ----------------->
+
+```{r setup, include=FALSE}
+long_slug <- "2024-05-08_tlg_catalog_webr"
+# renv::use(lockfile = "renv.lock")
+```
+
+<!--------------- post begins here ----------------->
+
+[![](tlg_catalog.png){fig-align="center" width="220"}](https://insightsengineering.github.io/tlg-catalog)
+
+[TLG Catalog website](https://insightsengineering.github.io/tlg-catalog)
+
+# Introducing WebR to TLG Catalog: A Game Changer for Interactive Learning
+
+I'm thrilled to announce a major update to the [TLG Catalog](https://insightsengineering.github.io/tlg-catalog). We have integrated [`WebR`](https://docs.r-wasm.org/webr/latest/) to the website transforming the way you interact with the R code. You don't have to copy and run code separately, open containers and/or install packages. With WebR, you can now interact with the TLG source R code directly in your browser, making learning more interactive and enjoyable than ever before!
+
+![](recording_static.gif){fig-align="center"}
+
+## What is `WebR`?
+
+> WebR makes it possible to run R code in the browser without the need for an R server to execute the code: the R interpreter runs directly on the user’s machine.
+
+_Source: [WebR documentation](https://quarto-webr.thecoatlessprofessor.com/)_
+
+In short, [`WebR`](https://docs.r-wasm.org/webr/latest/) is a project that aims to port R into [`WebAssembly`](https://webassembly.org/) (WASM) which then allows to run compiled code in the website. A special thanks to George Stagg from Posit for making this integration possible. While [`WebR`](https://docs.r-wasm.org/webr/latest/) is still in active development, a significant progress had been made recently increasing its robustness and efficiency.
+
+However, it's important to note a limitation: not all packages are compatible with [`WebR`](https://docs.r-wasm.org/webr/latest/). A package must be compiled for WebAssembly to be used with [`WebR`](https://docs.r-wasm.org/webr/latest/). Fortunately, there's a dedicated [WebR binary R package repository](https://repo.r-wasm.org) hosting close to 20,000 packages. For packages not yet available, you can utilize a dedicated [GitHub Actions workflow](https://github.com/r-wasm/actions) to build them yourself, or use [r-universe](https://ropensci.org/blog/2023/11/17/runiverse-wasm/).
```

Comment:
Could you elaborate what is the difference between r-wasm/actions and runiverse-wasm?
Are these two different approaches to integrate new packages into webassembly?